toolchain/check/handle_binding_pattern.cpp

```diff
@@ -20,6 +20,9 @@ static auto HandleAnyBindingPattern(Context& context, Parse::NodeId node_id,
 
   // TODO: Handle `_` bindings.
 
+  SemIR::ExprRegionId type_expr_region_id =
+      context.EndSubpatternAsExpr(cast_type_inst_id);
```

Comment:
Is this ending the subpattern started in `HandleIntroducer` for a let/var? Could we add a comment explaining where the subpattern started or somehow make it more explicit?